src/programs/azimuthal_average/azimuthal_average.cpp

```diff
@@ -0,0 +1,1534 @@
+	bool phase_flip_only = my_current_job.arguments[11].ReturnBoolArgument();
+	int max_iterations = my_current_job.arguments[12].ReturnIntegerArgument();
+	float minimum_shift_in_angstroms = my_current_job.arguments[13].ReturnFloatArgument();
+	float maximum_shift_in_angstroms = my_current_job.arguments[14].ReturnFloatArgument();
+	float termination_threshold_in_angstroms = my_current_job.arguments[15].ReturnFloatArgument();
+	float psi_min = my_current_job.arguments[16].ReturnFloatArgument();
+	float psi_max = my_current_job.arguments[17].ReturnFloatArgument();
+	float psi_step = my_current_job.arguments[18].ReturnFloatArgument();
+	float termination_threshold_in_degrees = my_current_job.arguments[19].ReturnFloatArgument();
+	float bfactor_in_angstroms = my_current_job.arguments[20].ReturnFloatArgument();
+	bool should_mask_central_cross = my_current_job.arguments[21].ReturnBoolArgument();
+	int horizontal_mask_size = my_current_job.arguments[22].ReturnIntegerArgument();
+	int vertical_mask_size = my_current_job.arguments[23].ReturnIntegerArgument();
+	float exposure_per_frame = my_current_job.arguments[24].ReturnFloatArgument();
+	int number_of_frames_for_running_average = my_current_job.arguments[25].ReturnIntegerArgument();
+	int max_threads = my_current_job.arguments[26].ReturnIntegerArgument();
+
+	// The Files
+	ImageFile my_input_file(input_filename, false);
+	NumericTextFile *input_text;
+	long number_of_input_images = my_input_file.ReturnNumberOfSlices();
+
+	if (input_ctf_values_from_text_file == true)
+	{
+		input_text = new NumericTextFile(text_filename, OPEN_TO_READ);
+		if (input_text->number_of_lines != number_of_input_images)
+		{
+			SendError("Error: Number of lines in defocus text file != number of images!");
+			DEBUG_ABORT;
+		}
+
+		if (input_text->records_per_line != 3 && input_text->records_per_line != 4)
+		{
+			SendError("Error: Expect 3 or 4 records per line in defocus text file");
+			DEBUG_ABORT;
+		}
+	}
+
+	// CTF object
+	// temporary array to hold CTF parameters read from file
+	float temp_float[5];
+	// allocate 2D array for CTF
+	int ncols_ctf = 11;
+	// dynamically create array of pointers of size number_of_images (rows)
+	float** ctf_parameters = new float*[number_of_input_images];
+	// dynamically allocate memory of size 11 for each row (columns)
+	for (int i = 0; i < number_of_input_images; i++)
+	{
+		ctf_parameters[i] = new float[ncols_ctf];
+	}
+
+	// assign values to allocated memory
+	for (int i = 0; i < number_of_input_images; i++)
+	{
+		for (int j = 0; j < ncols_ctf; j++)
+		{
+			ctf_parameters[i][j] = 0.0;
+		}
+	}
+
+	current_ctf.Init(acceleration_voltage, spherical_aberration, amplitude_contrast, defocus_1, defocus_2, astigmatism_angle, 0.0, 0.5, 0.0, pixel_size, additional_phase_shift);
+
+	// user input already
+	//float input_pixel_size = my_input_file.ReturnPixelSize();
+
+	// Profiling
+	wxDateTime	overall_start = wxDateTime::Now();
```

Comment:
The StopWatch class in cisTEM is trivial to use and will be more flexible than this approach.